brick-editor.js

```diff
@@ -63,19 +75,43 @@ function addBlock(word) {
 
     // initialize dictionary
     var blockDict = {
-        'var': "var variableName = value;",
-        'if': "if (i == true) {\n" + tabs + "\t" + "// do something \n" + tabs + "}",
-        'if-else': "if (i == true) {\n" + tabs + "\t" + "// do something \n" + tabs + "} else {\n" + tabs + "\t" + "// do something \n" + tabs + "}",
-        'for': "for (var i = 0; i < value; i++){\n" + tabs + "\t // do something \n" + tabs + "}",
-        'while': "while (i < 10) {\n" + tabs + "\t" + "// do something \n" + tabs + "}",
-        'function': "function name(parameters) {\n" + tabs + "\t // do something \n" + tabs + "\t return value;\n" + tabs + "}",
+        'VARIABLE': "var variableName = value;",
+        'IF': "if (i == true) {\n" + tabs + "\t" + "// do something \n" + tabs + "}",
+        'IF-ELSE': "if (i == true) {\n" + tabs + "\t" + "// do something \n" + tabs + "} else {\n" + tabs + "\t" + "// do something \n" + tabs + "}",
+        'FOR': "for (var i = 0; i < value; i++){\n" + tabs + "\t // do something \n" + tabs + "}",
+        'WHILE': "while (i < 10) {\n" + tabs + "\t" + "// do something \n" + tabs + "}",
+        'FUNCTION': "function name(parameters) {\n" + tabs + "\t // do something \n" + tabs + "\t return value;\n" + tabs + "}",
     };
```

Comment:
Instead, the better way to do this is to derive the keys from this dictionary. This might mean changing this from a dictionary to a nested list (so the key-order is preserved).

For the colors, you might pre-define some number of colors and pick from those. Alternately, you might hard-code the colors into the dictionary as well. I can see the latter approach being useful if the buttons will be in groups (ie. control flow vs. functions).

This means your "dictionary" might end up being a nested list like:

```
var blockDict = [
    ['var', 'var variableName = value;', '#ff3399'],
    // ...
];
```

Reply:
Or actually, even better would be to make this a full JSON object with nested dictionaries:

```javascript
var blockDict = [
    {
        "block-name":"var",
        "code":"var variableName = value;",
        "button-color":"#ff3399"
    },
    // ...
];
```

Reply:
is it better practice to use hexadecimal instead of the english color names?